pkg/platform/kube/functionres/lazy.go

```diff
@@ -1484,6 +1484,30 @@ func (lc *lazyClient) ensureServicePortsExist(to []v1.ServicePort, from []v1.Ser
 	return to
 }
 
+func (lc *lazyClient) GetCronTriggerInvocationURL(function *nuclioio.NuclioFunction,
+	resources Resources,
+	namespace string) (string, error) {
+
+	// if an ingress exists, use it
+	for _, ingress := range functionconfig.GetIngressesFromTriggers(function.Spec.Triggers) {
```

Comment:
the opposite, use the service first, you don't want to resolve an external DNS to know you are triggering a function that available within your cluster. it is a needlessly hop with overhead.

What you can do -
Provide a request header / invocation mode / etc that forces you to use the external ingress (for debug/testings only). Other than that, keep network within cluster if/when possible

Reply:
I removed the invocation by ingress, as it is not necessary (thought it would fix test failure to resolve function address on minikube tests, but it didn't)